warehouse/upload.go

```diff
@@ -396,15 +396,34 @@ func (job *UploadJobT) run() (err error) {
 				job.setStagingFilesStatus(job.stagingFiles, warehouseutils.StagingFileFailedState)
 				break
 			}
-
 			err = job.setLoadFileIDs(startLoadFileID, endLoadFileID)
 			if err != nil {
 				break
 			}
 
 			job.matchRowsInStagingAndLoadFiles()
 			job.recordLoadFileGenerationTimeStat(startLoadFileID, endLoadFileID)
-
+			if job.warehouse.Type == "S3_DATALAKE" {
+				for tableName := range job.upload.UploadSchema {
+					loadFiles := job.GetLoadFilesMetadata(warehouseutils.GetLoadFilesOptionsT{
+						Table:   tableName,
+						StartID: startLoadFileID,
+						EndID:   endLoadFileID,
+					})
+					// This is best done every 100 files, since it's a batch request for updates in Glue
+					partitionBatchSize := 99
+					timeWindowFormat, _ := job.warehouse.Destination.Config["timeWindowFormat"].(string)
```

Comment:
We need to handle the scenarios where `timeWindowFormat` is not  provided in config